seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/JobExecutionEnvironment.java

```diff
@@ -64,6 +83,82 @@ protected MultipleTableJobConfigParser getJobConfigParser() {
                 jobFilePath, idGenerator, jobConfig, commonPluginJars, isStartWithSavePoint);
     }
 
+    @Override
+    protected LogicalDag getLogicalDag() {
+        ImmutablePair<List<Action>, Set<URL>> immutablePair = getJobConfigParser().parse();
+        actions.addAll(immutablePair.getLeft());
+        // Enable upload connector jar package to engine server, automatically upload connector Jar
+        // packages
+        // and dependent third-party Jar packages to the server before job execution.
+        // Enabling this configuration does not require the server to hold all connector Jar
+        // packages.
+        boolean enableUploadConnectorJarPackage =
+                seaTunnelConfig.getEngineConfig().getConnectorJarStorageConfig().getEnable();
+        if (enableUploadConnectorJarPackage == true) {
+            Set<ConnectorJarIdentifier> commonJarIdentifiers =
+                    connectorPackageClient.uploadCommonPluginJars(
+                            Long.parseLong(jobConfig.getJobContext().getJobId()), commonPluginJars);
```

Comment:
Do we need to check if the jar already exist in Master?

Reply:
If we send a request from the client to the server to determine whether the connector Jar package exists, then if the Jar package exists, a separate request API is required under the shared storage strategy to increase the Jar package reference count of the connector JarRefCounters in the shared storage mode. Do I need to change to this method?

Reply:
If we send a  request from the client to the server to check for the existence of the connector Jar package, and if the Jar package does exist, we need a separate API request to increment the reference count for the connector Jar package in shared connector jar storage mode. Should we switch to this approach?

Reply:
You can add todo here, this is not needed, but this can help to reduce the request size, since most of the Job may have the same connector jars.

Reply:
Thank you very much for your suggestion. I have added todo, PTAL!

Reply:
This feature can be done in server side. When client upload jar to server, server can do check for jar already existed or not. If existed, just skip save jar into server, finally return successed to client. So you don't need to send another check request, all be done in one request.

Reply:
I understand your point and have implemented the the check of Jar package on the server side. PTAL!